### PR TITLE
NIFI-5068 Script to automate parts of RC validation

### PR DESCRIPTION
**Pull request by ottobackwards**
Currently this script will 
- download and validate the hashes and signing
- unpack the zip.
- pull the commit and diff with the unpacked sources
- build and run tests
- validate that nifi starts

Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by joewitt** [comment]
i dont think we should commit this into the codebase.  It could be fine to publish the bash script in the wiki page associated with release guide/validation.  We must also be clear this is good for things which are easily automated such as verification of sigs/hashes/build but for other things like actually testing nifi and the hard work of L&N review its an eyes-on game.

**Comment by ottobackwards** [comment]
ok @joewitt, but did it work? ;)

**Comment by ottobackwards** [comment]
How would we go about getting it on the wiki page?

File: dev-utilities/release-utilities/nifi-rc-check [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+shopt -s nocasematch
+
+function print_help {
+  echo " "
+  echo "usage: ${0}"
+  echo "    -v/--version=<version>   The version of the Nifi release. [Required]"
```
Comment (alopresto) [inline dev-utilities/release-utilities/nifi-rc-check]: s/Nifi/NiFi/

File: dev-utilities/release-utilities/nifi-rc-check [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+print_section_item "importing nifi keys"
+
+if ! gpg --import KEYS ; then
+  print_error "failed to import KEYS"
+  exit 1
+fi
+
+print_section_item "Verifying Nifi Assembly"
+if ! gpg --verify ./"nifi-$NIFI_VERSION-source-release.zip.asc" "nifi-$NIFI_VERSION-source-release.zip" ; then
+  print_error "failed to verify Nifi Assembly"
```
Comment (alopresto) [inline dev-utilities/release-utilities/nifi-rc-check]: This should use `gpg --verify -v ...` to ensure that the internal digest algorithm being used for the signature is SHA-512.

File: dev-utilities/release-utilities/nifi-rc-check [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+  exit 1
+fi
+
+LOCAL_SHA1="$(sha1sum ${LOCAL_ASSEMBLY} | cut -d' ' -f1)";
```
Comment (alopresto) [inline dev-utilities/release-utilities/nifi-rc-check]: I do not believe `sha1sum` is available on Mac OS X; use `shasum -a 1` instead.

File: dev-utilities/release-utilities/nifi-rc-check [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+fi
+
+print_section_header "Build and Test"
+echo ""
+echo ""
+read -p " Build Nifi and run contrib check? [yN]" -n 1 -r
+echo
+if [[ $REPLY =~ ^[Yy]$ ]]; then
+  print_section_item "Performing build with contrib-check"
+  if ! mvn -f nifi-${NIFI_VERSION}/pom.xml clean install -Pcontrib-check ; then
```
Comment (alopresto) [inline dev-utilities/release-utilities/nifi-rc-check]: I think this should include `-Pinclude-grpc` for verifying the build during a release vote.

**Comment by ottobackwards** [comment]
@alopresto I have made the suggested changes.  Unfortunately the build has been pulled, so the script won't run until next release.

**Comment by joewitt** [comment]
@ottobackwards i'd recommend just moving this to be a bash script made available on the wiki and close the PR.

You could 
1. Propose an edit to the release process to make this available as an alternative way for folks who want to validate the RC (as an alternative to the current path)
2. Put this bash script on the wiki and link to it from #1.

Thanks

**Comment by ottobackwards** [comment]
@joewitt I agree.  I'll see if I can edit the wiki ( if I have rights )

**Comment by ottobackwards** [comment]
I do not have rights to create a page in the space from what I can tell

**Comment by joewitt** [comment]
should have them now

**Comment by ottobackwards** [comment]
I sent the email, I'll close this PR with the results of that when they are set

**Comment by ottobackwards** [comment]
closing, not much demand I guess.

**Pull request closed without merging**